client/js/app.js

```diff
@@ -38,8 +68,114 @@ function setLoading(loading) {
+        td1.append(key.replace(/_/g, " "));
+        tr.appendChild(td1);
+        let td2 = document.createElement("td");
+        td2.setAttribute("class", "mdl-data-table__cell--non-numeric");
+        td2.append(data[key].type);
+        tr.appendChild(td2);
+        table.appendChild(tr);
+    });
+}
+
+function getColumnData(dataset) {
+    loadAdditionalData("columns", "POST", {dataset: dataset}, getColumnDataSuccess, () => {
+        // TODO handle error
+        setLoading(false)
+    });
+}
+
+function onExampleSelected(example) {
+    console.log(example.textContent);
+    let ul = document.getElementById("dataset-list");
+    for (var i = 0; i < ul.children.length; i++) {
+        if (ul.children[i].dataset.selected === 'true') {
+            var dataset = ul.children[i].dataset.val;
+            break;
+        }
+    }
+    generateGraph(dataset, example.textContent)
+}
+
+function getExampleDataSuccess(response) {
+    let data = response.examples;
+    let keys = Object.keys(data);
+    let table = document.getElementById('example_body');
+    while (table.firstChild) {
+        table.removeChild(table.firstChild);
+    }
+    data.forEach(d => {
+        let tr = document.createElement("tr");
+        let td = document.createElement("td");
+        td.setAttribute("class", "mdl-data-table__cell--non-numeric");
+        td.append(d.input);
+        tr.onclick = function () {
+            onExampleSelected(this);
+        };
+        tr.appendChild(td);
+        table.appendChild(tr);
+    })
+}
+
+function getExamplesData(dataset) {
+    loadAdditionalData("examples", "POST", {dataset: dataset}, getExampleDataSuccess, () => {
+        // TODO handle error
+        setLoading(false)
+    });
+}
+
+function getAdditionalData(selectedDataset) {
+    getColumnData(selectedDataset);
+    getExamplesData(selectedDataset);
+}
 
+function onDatasetSelected(dataset) {
+    if (dataset.dataset.selected === 'false') {
+        let ul = document.getElementById("dataset-list");
+        for (let i = 0; i < ul.children.length; i++) {
+            ul.children[i].dataset.selected = (dataset.dataset.val === ul.children[i].dataset.val);
+        }
+        getAdditionalData(dataset.dataset.val);
+    }
+}
+
+function loadDatasetSuccess(data) {
+    let ul = document.getElementById("dataset-list");
+    data.forEach(d => {
+        let li = document.createElement("li");
+        li.setAttribute("class", "mdl-menu__item");
+        li.append(d.display_name);
+        li.dataset.val = d.table_name;
+        li.dataset.selected = false;
+        li.onclick = function () {
+            onDatasetSelected(this);
+        };
+        ul.appendChild(li);
+    });
+    ul.firstElementChild.dataset.selected = true;
+    getAdditionalData(ul.firstElementChild.dataset.val);
+}
+
+function loadDockData() {
+    loadDatasets("datasets", "GET", loadDatasetSuccess, () => {
+        // TODO handle error
+        setLoading(false)
```

Comment:
this should be always in the same function where `setLoading(true)` is called

Reply:
I dont understand your point. loadDockData is being called on window.onload